File: Server/src/main/java/com/example/cua_chat_app/entity/notification/Notification.java
package com.example.cua_chat_app.entity.notification;

import com.example.cua_chat_app.entity.message.MessageType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Notification {

    private String chatId;
    private String content;
    private String senderId;
    private String receiverId;
    private String chatRoomName;
    private MessageType messageType;
    private NotificationType type;
    private byte[] media;
}

File: client/src/app/pages/main/models/notification.ts
export interface Notification {
    chatRoomId?: string;
    content?: string;
    senderId?: string;
    receiverId?: string;
    messageType?: 'TEXT' | 'IMAGE' | 'VIDEO' | 'AUDIO';
    type?: 'SEEN' | 'MESSAGE' | 'IMAGE' | 'VIDEO' | 'AUDIO';
    chatRoomName?: string;
    media?: Array<string>;
  }